### DISCUSSION TITLE
Cyceling through Chain Skills

### DISCUSSION THREAD

**Discussion by SaculRennorb** [HardstuckGuild/GW2-Tooltips.js#5] [General]
## Should we actually have the skill cycle on chained skills / traits with attached skills or should we always display all skills?

mightyteapot: Autoattack chains seem to not display facts except for the first attack
mightyteapot: -> it seems that right clicking cycles though the auto chain but I'm not sure if that's desirable
mightyteapot: -> that seems to apply to all chain skills

A collection of pro and contra points:

### Pros (do the cycling):
- Takes up less space.
  - If we want to show all kills in bundles  we will need to think of a different solution for that.
- Focus is always on one box, declutters the screen.
- The current implementation works similarly, but only ever shows one tooltip and not the minified versions of the remaining ones.

### Cons (always show all):
- The game always shows all (i think).
- Very rarely makes it hard to think about a skill where multiple blocks would be interesting.
- Code complexity (relatively minor).
- It prevents you from using the normal context menu, which in turn prevents you from opening the wiki page in a new tab the 'usual' way. I didn't consider this, as a middle mouse click does the same thing and i have been using that since i can remember. 

---
Another option to make this more user friendly might also be to just place the hint differently. Currently it only shows a small text box that is easily overlooked, which indicates how to cycle the tooltips. On mobile this cannot reliably show up at all
![grafik](https://github.com/HardstuckGuild/Tooltips.js/assets/18741506/fc488aaa-5c6b-4812-a045-6d40203a99b8)

So maybe just moving the hit into the tooltip itself could be the solution:
![grafik](https://github.com/HardstuckGuild/Tooltips.js/assets/18741506/d2273b2f-71e8-4357-9c02-027922a6782a)

**Comment by Wallaka** [comment]
I will add my thoughts here too:

### **Pros (do the cycling)**:

- On mobile the experience is better handled, because the tooltip won't go out of the screen easily, if you display all 3 autos (if you have 3 chains), it can be quite difficult to have a good experience on mobile


### **Cons (always show all)**:

- I am afraid that 95% of the user won't ever see this mechanism as it is maybe not widely spread thing on the internet, it is like a pro mechanism let's say
- Also, if you do this, the users can't do right click, `open in new tab` etc, I know that might not be a very used feature, but I think some people are doing it, like if they want the wiki page of it

**Reply by SaculRennorb** [reply]
Regarding the rclick -> open in new tab;
Its true , i never even thought of it. I haven't used that for years now since clicking the middle mouse button does the same thing with fewer menus, so i have been using that since forever instead.

**Comment by SaculRennorb** [comment]
Just a short update: 

For now this mostly settled on keeping it cycling through collapsed tooltips, as other options get unwieldy when showing a lot of skills, for example the bundle display.
We might be able to resolve this if we chose to change how bundles and transformations are displayed.

I will however keep this open, as we are still looking for better solutions.
